core/src/test/java/hudson/RemoveWindowsDirectoryJunctionTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *
+ */
+package hudson;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+import java.io.File;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.Issue;
+
+public class RemoveWindowsDirectoryJunctionTest {
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+
+    @Before
+    public void windowsOnly() {
+       assumeTrue(File.pathSeparatorChar==';');
+    }
+
+    @Test
+    @Issue("JENKINS-2995")
+    public void testJunctionIsRemovedButNotContents() throws Exception {
+        TemporaryFolder dir1 = new TemporaryFolder(tmp.getRoot());
+        dir1.create();
+        File dir1Root = dir1.getRoot();
+        File f1 = dir1.newFile();
+        File j1 = makeJunction(tmp.getRoot(), dir1Root);
```

Comment:
create directories under tmp instead to make sure cleanup happens correctly.  (`tmp.newFolder(...)`)